app/init/push_notifications.ts

```diff
@@ -148,9 +149,6 @@ class PushNotifications {
                     serverUrl,
                 };
 
-                // Dismiss the screen if it's already visible or else it blocks the navigation
-                await dismissOverlay(screen);
-
```

Comment:
the event above, already takes care of dismissing any previous in-app notification overlay, this second one was causing an issue where the navigation library did not set the props, thus the reason why `notification` was undefined.